programs/did-registry/src/lib.rs

```diff
@@ -17,21 +34,25 @@ impl Id for SolDID {
 #[program]
 pub mod did_registry {
     use super::*;
+    use crate::eth_signing::validate_eth_signature;
+    use sol_did::integrations::is_authority;
 
+    /// Create an empty DID registry for a given solana key
     pub fn create_key_registry(ctx: Context<CreateKeyRegistry>, _bump: u8) -> Result<()> {
         ctx.accounts.registry.authority = ctx.accounts.authority.key();
         Ok(())
     }
 
+    /// Add a DID to an authority's registry
     pub fn register_did(ctx: Context<RegisterDid>, _did_bump: u8) -> Result<()> {
         // ensure the authority is an authority on the did account
         // note, anchor has already verified the constraint that did_account
         // is the account for the did.
-        sol_did::is_authority(
+        is_authority(
             &ctx.accounts.did_account.to_account_info(),
-            &[], // the authority must be a direct authority on the DID
-            &ctx.accounts.authority.key(),
+            None, // the authority must be a direct authority on the DID
             &[],
+            ctx.accounts.authority.key().as_ref(),
```

Comment:
I assume this is the new `key: &[u8]` interface. I'm curious why you don't need to use as_bytes(). Is that some magic `Into` Trait action?

Also any reason you prefer `as_ref()` to `&`?

Reply:
I think Pubkey only has to_bytes which returns [u8:32]. So we would also need to add an as_slice() to that.

But in general, I feel like the "rustiest" approach is to prefer traits like AsRef, or Into, rather than type-specific functions like as_bytes, to allow the compiler to infer the required type based on the code being called.

Imagine Pubkey has two impls of AsRef: `AsRef<[u8]>` and `AsRef<str>`.

The client code you are calling can then require either `key: &[u8]` or `key: &str` and `authority.key().as_ref()` would work with both. as_ref is therefore "pass this struct (as a reference) in whatever format the calling code needs it". The more it is used, the more powerful it becomes.